frontend/src/components/ExperimentsTableConfigurator.jsx

```diff
@@ -51,31 +52,51 @@ class ExperimentsTableConfigurator extends React.Component {
         : hiddenLogKeys.filter((vk) => vk !== event.target.name);
 
       this.props.onTableColumnsVisibilityUpdate(
-        this.props.project.id, nextHiddenLogKeys, hiddenArgKeys);
+        this.props.project.id, nextHiddenLogKeys, hiddenArgKeys, isGrouped);
     } else {
       const nextHiddenArgKeys = !event.target.checked
         ? hiddenArgKeys.concat(event.target.name)
         : hiddenArgKeys.filter((vk) => vk !== event.target.name);
 
       this.props.onTableColumnsVisibilityUpdate(
-        this.props.project.id, hiddenLogKeys, nextHiddenArgKeys);
+        this.props.project.id, hiddenLogKeys, nextHiddenArgKeys, isGrouped);
     }
   }
 
+  handleIsGrouped(event) {
+    const { tableState } = this.props.projectConfig;
+
+    this.props.onTableColumnsVisibilityUpdate(
+      this.props.project.id, tableState.hiddenLogKeys, tableState.hiddenArgKeys,
+      event.target.checked
+    );
+  }
+
   render() {
     const { stats, projectConfig } = this.props;
     const { logKeys, argKeys } = stats;
     const { tableState } = projectConfig;
     const {
       hiddenLogKeys = [],
-      hiddenArgKeys = []
+      hiddenArgKeys = [],
+      isGrouped = false
     } = tableState;
 
     return (
       <div>
-        <Button color="secondary" className="my-2" onClick={this.handleModalShow}>
-          Table settings
-        </Button>
+        <div className="form-inline">
+          <Button color="secondary" className="my-2" onClick={this.handleModalShow}>
+            Table settings
+          </Button>
+          <Check
+            type="checkbox"
+            className="ml-2"
+            checked={isGrouped}
+            onChange={(e) => this.handleIsGrouped(e)}
```

Comment:
Such an event handler coding is not recommended.
First define `this.handleIsGrouped = this. handleIsGrouped.bind(this)` in the constructor and rewrite it with `onChange={this.handleIsGrouped}`.

Reply:
thx, followed